### ISSUE TITLE
Migrate to MapLibre GL NodeJS bindings

### ISSUE THREAD

**Issue by brendan-ward** [consbio/mbgl-renderer#95]
Modern NodeJS bindings are [now available from MapLibre](https://www.npmjs.com/package/maplibre-gl) ([docs](https://github.com/maplibre/maplibre-gl-native/tree/main/platform/node)).  There is still [active development](https://github.com/maplibre/maplibre-gl-native/pull/535) around compiling and packaging which may impact our usage here.

There is some work in progress toward integrating MapLibre GL Native here in the `test_maplibre_gl_native` branch (though outdated) and in #94.

Because the lack of maintained NodeJS bindings from Mapbox caused this project to stall out for quite some time, migrating to MapLibre is the next big step to active development on `mbgl-renderer` and needs to be done before other major work.

Proposed steps:
- [ ] replace Mapbox Node JS library with MapLibre Node JS library and set minimum required version of NodeJS to 14
- [ ] update dependencies in Docker containers
- [ ] update JS dependencies only as necessary for NodeJS 14+ / MapLibre
- [ ] update render tests as needed, manually verifying changes
- [ ] update docs

In theory, most of our tests should still work with minimal changes, including those that use Mapbox sources / tilesets, because we provide the routing to Mapbox API endpoints here.  It should be unaffected by [upcoming changes to MapLibre GL Native](https://github.com/maplibre/maplibre-gl-native/issues/349) that remove special handling of Mapbox tokens / API endpoints in the core.

As folks have surely noticed, I haven't been doing active development on this project lately.  This roadmap is not a commitment on my part to complete these steps in a timely fashion, only to lay out what I think needs to be done in case anyone wants to get started on this first.  PRs very welcome to help with this, just please keep them small, focused, and easy to review.
